packages/@orbit/jsonapi/src/jsonapi-request-processor.ts

```diff
@@ -0,0 +1,256 @@
+import Orbit, {
+  ClientError,
+  KeyMap,
+  NetworkError,
+  Operation,
+  Query,
+  Record,
+  Schema,
+  ServerError,
+  Transform,
+} from '@orbit/data';
+import { InvalidServerResponse } from './lib/exceptions';
+import { TransformRecordRequest } from './lib/transform-requests';
+import { deepGet, deepMerge, toArray } from '@orbit/utils';
+import { RecordDocument } from './record-document';
+import {
+  RequestOptions,
+  buildFetchSettings
+} from './lib/request-settings';
+import JSONAPIURLBuilder, { JSONAPIURLBuilderSettings } from './jsonapi-url-builder';
+import { JSONAPISerializer, JSONAPISerializerSettings } from './jsonapi-serializer';
+const { assert, deprecate } = Orbit;
+
+export interface FetchSettings {
+  headers?: object;
+  method?: string;
+  json?: object;
+  body?: string;
+  params?: any;
+  timeout?: number;
+  credentials?: string;
+  cache?: string;
+  redirect?: string;
+  referrer?: string;
+  referrerPolicy?: string;
+  integrity?: string;
+}
+
+
+export interface JSONAPIRequestProcessorSettings {
+  sourceName: string;
+  SerializerClass?: (new (settings: JSONAPISerializerSettings) => JSONAPISerializer);
+  URLBuilderClass?: (new (settings: JSONAPIURLBuilderSettings) => JSONAPIURLBuilder);
+  namespace?: string;
+  host?: string;
+  defaultFetchHeaders?: object;
+  defaultFetchTimeout?: number;
+  defaultFetchSettings?: FetchSettings;
+  allowedContentTypes?: string[];
+  schema: Schema;
+  keyMap: KeyMap;
+}
+
+export default class JSONAPIRequestProcessor {
+  sourceName: string;
+  SerializerClass?: (new (settings: JSONAPISerializerSettings) => JSONAPISerializer);
+  serializer: JSONAPISerializer;
+  URLBuilderClass?: (new (settings: JSONAPIURLBuilderSettings) => JSONAPIURLBuilder);
+  urlBuilder: JSONAPIURLBuilder;
+  allowedContentTypes: string[];
+  defaultFetchSettings: FetchSettings;
+  host: string;
+  namespace: string;
```

Comment:
Same for `host` and `namespace` - since they are passed directly into `URLBuilderClass`, does the `JSONAPIRequestProcessor` need to hold onto a reference?

Reply:
Same as above. I'll remove.